recipes/premake/5.x/conanfile.py

```diff
@@ -0,0 +1,52 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment, MSBuild
+import os
+
+
+class PremakeConan(ConanFile):
+    name = "premake"
+    version = "5.0.0-alpha14"
+    topics = ("conan", "premake", "build", "build-systems")
+    description = "Describe your software project just once, using Premake's simple and easy to read syntax, and build it everywhere"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://premake.github.io/"
+    license = "BSD-3-Clause"
+    settings = "os_build", "arch_build", "compiler"
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    @property
+    def _platform(self):
+        return {'Windows': 'vs2017',
+                'Linux': 'gmake.unix',
+                'Macos': 'gmake.macosx'}.get(str(self.settings.os_build))
+
+    def build(self):
+        with tools.chdir(os.path.join(self._source_subfolder, 'build', self._platform)):
+            if self.settings.os_build == 'Windows':
```

Comment:
This is only valid when `compiler == "Visual Studio"`
I think there is another build directory when using mingw.

Reply:
Even when handling MinGW explicitly here the build fails eventually
https://github.com/bincrafters/conan-center-index/commit/80ff869bde76fd75844357ebf1975cc8f7b9a270

For the moment I will raise an `ConanInvalidConfiguration` when trying to build with MinGW

Reply:
Yeah, please, just add a section for mingw and we are good!